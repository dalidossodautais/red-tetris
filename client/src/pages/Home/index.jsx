import React from "react";
import styled from "styled-components";
import Rooms from "./Rooms";
import Scores from "./Scores";

const BodyContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-evenly;
  width: 100%;
`;

export default function Home() {
  return (
    <BodyContainer>
      <Rooms />
      <Scores />
    </BodyContainer>
  );
}
