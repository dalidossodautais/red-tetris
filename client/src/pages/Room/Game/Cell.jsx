import React from "react";
import styled from "styled-components";

function darken(colorPart) {
  return (colorPart * 3) / 4;
}
function lighten(colorPart) {
  return (colorPart * 5) / 4;
}

const colors = (() => {
  const basics = {
    I /* Cyan */: "rgb(0,184,212)",
    J /* Pink */: "rgb(233,30,99)",
    L /* Orange */: "rgb(255,109,0)",
    O /* Yellow */: "rgb(255,214,0)",
    S /* Teal */: "rgb(0,150,136)",
    T /* Deep Purtle */: "rgb(103,58,183)",
    Z /* Red */: "rgb(244,67,54)",
    full: "rgb(64,64,64)",
    shadow: "rgb(192,192,192)",
    null: "rgb(220,220,220)",
  };
  const merged = {};
  Object.entries(basics).forEach(([key, basic]) => {
    const parted = basic.match(/\d+/g);
    merged[key] = {
      basic,
      dark: `rgb(${darken(parted[0])},${darken(parted[1])},${darken(
        parted[2]
      )})`,
      light: `rgb(${lighten(parted[0])},${lighten(parted[1])},${lighten(
        parted[2]
      )})`,
    };
  });
  return merged;
})();

const StyledCell = styled.div.attrs((props) => {
  if (!props.transparent)
    return {
      backgroundColor: props.basic,
      borderBottomColor: props.dark,
      borderLeftColor: props.light,
      borderRightColor: props.dark,
      borderTopColor: props.light,
    };
  return {
    backgroundColor: "transparent",
    borderBottomColor: "transparent",
    borderLeftColor: "transparent",
    borderRightColor: "transparent",
    borderTopColor: "transparent",
  };
})`
  background-color: ${(props) => props.backgroundColor};
  border-bottom-color: ${(props) => props.borderBottomColor};
  border-left-color: ${(props) => props.borderLeftColor};
  border-right-color: ${(props) => props.borderRightColor};
  border-style: solid;
  border-width: ${(props) => props.size * 0.05}px;
  border-top-color: ${(props) => props.borderTopColor};
  height: ${(props) => props.size * 0.9}px;
  opacity: ${(props) => props.opacity};
  width: ${(props) => props.size * 0.9}px;
`;

export default function Cell({ cell = null, empty, opacity, size }) {
  const { basic, dark, light } = colors[cell];
  return (
    <StyledCell
      basic={basic}
      dark={dark}
      light={light}
      opacity={opacity}
      size={size}
      transparent={empty && !cell}
    />
  );
}
