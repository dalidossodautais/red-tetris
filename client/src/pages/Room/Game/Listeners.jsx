import { useCallback, useEffect, useRef } from "react";
import { playGame } from "../../../middleware";

function useListeners(key, action) {
  const timeout = useRef(null);
  const handleRepeatedKeys = useCallback(
    (time) => {
      timeout.current = setTimeout(() => {
        playGame(action);
        handleRepeatedKeys(50);
      }, time);
    },
    [action]
  );
  const downHandler = useCallback(
    (event) => {
      if (event.key === key && !timeout.current) {
        playGame(action);
        handleRepeatedKeys(400);
      }
    },
    [action, handleRepeatedKeys, key]
  );
  const upHandler = useCallback(
    (event) => {
      if (event.key === key) {
        clearTimeout(timeout.current);
        timeout.current = null;
      }
    },
    [key]
  );
  useEffect(() => {
    window.addEventListener("keydown", downHandler);
    window.addEventListener("keyup", upHandler);
    return () => {
      window.removeEventListener("keydown", downHandler);
      window.removeEventListener("keyup", upHandler);
      clearTimeout(timeout.current);
    };
  }, [downHandler, upHandler]);
}

export default function Listeners() {
  useListeners("ArrowDown", "down");
  useListeners("ArrowLeft", "left");
  useListeners("ArrowRight", "right");
  useListeners("ArrowUp", "rotate");
  useListeners(" ", "bottom");
  useListeners("0", "keep");
  return null;
}
