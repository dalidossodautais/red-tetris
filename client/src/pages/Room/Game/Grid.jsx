import React, { useContext } from "react";
import styled from "styled-components";
import { RoomContext } from "..";
import Cell from "./Cell";

const Container = styled.div`
  display: flex;
  flex-direction: column;
`;

const GridContainer = styled.div.attrs((props) => ({
  height: `${props.heightInVh}vh`,
}))`
  align-items: center;
  display: flex;
  flex-direction: column;
  height: ${(props) => props.height};
  justify-content: center;
`;

const LineContainer = styled.div`
  background-color: black;
  display: flex;
  flex-direction: row;
`;

const Text = styled.a.attrs((props) => ({
  color: props.theme.primaryTextColor,
  fontSize: `${(props.heightInVh * 8) / 10}vh`,
  height: `${props.heightInVh}vh`,
}))`
  color: ${(props) => props.color};
  font-size: ${(props) => props.fontSize};
  height: ${(props) => props.height};
  text-overflow: ellipsis;
`;

export default function Grid({ heightInVh, playerName }) {
  const {
    [`game_${playerName}`]: { grid, isPlaying, points } = {},
  } = useContext(RoomContext);
  const fontHeightInVh = heightInVh / 20;
  if (!grid) return null;
  const size = Math.floor(
    ((window.innerHeight / Math.max(grid.length, 4)) * heightInVh) / 100
  );
  return (
    <Container>
      <GridContainer heightInVh={`${heightInVh - fontHeightInVh}vh`}>
        {grid.map((line, index) => (
          <LineContainer key={index}>
            {line.map((cell, index) => (
              <Cell
                cell={cell}
                key={index}
                opacity={isPlaying ? 1 : 0.3}
                size={size}
              />
            ))}
          </LineContainer>
        ))}
      </GridContainer>
      <Text
        heightInVh={fontHeightInVh}
      >{`${playerName} with ${points} points`}</Text>
    </Container>
  );
}
