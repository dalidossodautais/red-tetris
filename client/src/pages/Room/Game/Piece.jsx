import React from "react";
import styled from "styled-components";
import Cell from "./Cell";

const Container = styled.div`
  display: flex;
  flex-direction: column;
`;

const GridContainer = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  height: ${(props) => props.height};
  justify-content: center;
`;

const LineContainer = styled.div`
  background-color: transparent;
  display: flex;
  flex-direction: row;
`;

export default function Piece({ grid = [], height }) {
  const size = Math.floor(
    ((window.innerHeight / Math.max(grid.length, 4)) * height) / 100
  );
  return (
    <Container>
      <GridContainer height={`${height}vh`}>
        {grid.map((line, index) => (
          <LineContainer key={index}>
            {line.map((cell, index) => (
              <Cell cell={cell} empty key={index} size={size} />
            ))}
          </LineContainer>
        ))}
      </GridContainer>
    </Container>
  );
}
