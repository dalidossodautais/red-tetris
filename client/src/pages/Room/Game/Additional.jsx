import React, { useContext } from "react";
import styled from "styled-components";
import Header from "../../../components/card/Header";
import Subtitle from "../../../components/texts/Subtitle";
import { RoomContext } from "..";
import Piece from "./Piece";

const Container = styled.div`
  background-color: #423434;
  border-radius: 5px;
  height: 100%;
  min-width: 400px;
  width: ${(props) => props.width};
`;

const GridsContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export default function Additional({ playerName, width }) {
  const { [`game_${playerName}`]: { keeped, next } = {} } = useContext(
    RoomContext
  );
  if (!next) return null;
  return (
    <Container width={width}>
      <Header iconName="gamepad" title="Additional" />
      <Subtitle textAlign="center">Keeped Piece</Subtitle>
      <Piece grid={keeped} height={15} />
      <Subtitle textAlign="center">Next Pieces</Subtitle>
      <GridsContainer>
        {next.map((piece, index) => (
          <Piece grid={piece} height={15} key={index} />
        ))}
      </GridsContainer>
    </Container>
  );
}
