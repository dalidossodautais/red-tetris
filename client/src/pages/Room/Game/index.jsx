import React, { useContext, useEffect, useMemo } from "react";
import styled from "styled-components";
import { RoomContext } from "..";
import { handleGames, unhandleGames } from "../../../middleware";
import Additional from "./Additional";
import Grid from "./Grid";
import Listeners from "./Listeners";

const Container = styled.div`
  align-items: flex-start;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-evenly;
  width: 100%;
`;

const SubContainer = styled.div`
  align-items: flex-start;
  display: flex;
  flex-direction: column;
  @media (max-width: ${(props) => 2.5 * props.elementWidth}px) {
    display: none;
  }
`;

export default function Game() {
  const { isPlayer, options, playerNames, setGame } = useContext(RoomContext);
  useEffect(() => {
    handleGames(setGame, playerNames);
    return () => {
      unhandleGames(playerNames);
    };
  }, [playerNames, setGame]);
  const mainHeightInVh = 85;
  const elementWidth = useMemo(
    () =>
      (((window.innerHeight / options.height) * mainHeightInVh) / 100) *
      options.width,
    [options.height, mainHeightInVh, options.width]
  );
  const [mainPlayerName, ...secondaryPlayerNames] = playerNames;
  return (
    <Container>
      {mainPlayerName && (
        <>
          {isPlayer && <Listeners />}
          <Grid heightInVh={mainHeightInVh} playerName={mainPlayerName} />
          <Additional playerName={mainPlayerName} width={`${elementWidth}px`} />
        </>
      )}
      <SubContainer elementWidth={elementWidth}>
        {secondaryPlayerNames.map((playerName, index) => (
          <Grid
            heightInVh={mainHeightInVh / 3}
            key={index}
            playerName={playerName}
          />
        ))}
      </SubContainer>
    </Container>
  );
}
