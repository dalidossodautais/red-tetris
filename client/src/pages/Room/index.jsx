import React, {
  createContext,
  useCallback,
  useContext,
  useEffect,
  useReducer,
} from "react";
import { handleRoom, unhandleRoom } from "../../middleware";
import { Context } from "../../Provider";
import Game from "./Game";
import Lobby from "./Lobby";

export const RoomContext = createContext({});

export default function Room() {
  const [room, setRoom] = useReducer(
    (room, newRoom) => ({ ...room, ...newRoom }),
    {}
  );
  const setGame = useCallback(
    (game, playerName) => setRoom({ [`game_${playerName}`]: game }),
    [setRoom]
  );
  const { username } = useContext(Context);
  useEffect(() => {
    handleRoom(setRoom, username);
    return () => {
      unhandleRoom();
    };
  }, [setRoom, username]);
  return (
    Object.keys(room).length > 0 && (
      <RoomContext.Provider value={{ ...room, setGame }}>
        {room.isRunning ? <Game /> : <Lobby />}
      </RoomContext.Provider>
    )
  );
}
