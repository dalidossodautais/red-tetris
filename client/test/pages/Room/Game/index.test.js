import React from "react";
import { render } from "@testing-library/react";
import { RoomContext } from "../../../../src/pages/Room";
import Game from "../../../../src/pages/Room/Game";

describe("Game", () => {
  it("Empty Render", () => {
    const component = render(
      <RoomContext.Provider value={{ playerNames: ["main"], options: [] }}>
        <Game />
      </RoomContext.Provider>
    );
    expect(component).toMatchSnapshot();
  });
  it("Filled Render", () => {
    const component = render(
      <RoomContext.Provider
        value={{
          isPlayer: true,
          game_main: { grid: [[null]], next: [[[null, "I", undefined]]] },
          playerNames: ["main", "secondary"],
          options: [{}],
        }}
      >
        <Game />
      </RoomContext.Provider>
    );
    expect(component).toMatchSnapshot();
  });
  it("Filled Render", () => {
    const component = render(
      <RoomContext.Provider
        value={{
          isPlayer: false,
          game_main: { grid: [[null]], isPlaying: true },
          game_secondary: { grid: [[null]] },
          playerNames: ["main", "secondary"],
          options: [{}],
        }}
      >
        <Game />
      </RoomContext.Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
