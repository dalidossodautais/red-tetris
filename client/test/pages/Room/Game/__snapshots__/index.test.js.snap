// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Empty Render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-crrsfI iVnQMr"
      >
        <div
          class="sc-dQppl hMUciP"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-crrsfI iVnQMr"
    >
      <div
        class="sc-dQppl hMUciP"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Game Filled Render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-crrsfI iVnQMr"
      >
        <div
          class="sc-pFZIQ gwVLIm"
        >
          <div
            class="sc-jrAGrp bNQYBx"
            height="80.75vhvh"
          >
            <div
              class="sc-kEjbxe kkVOZj"
            >
              <div
                class="sc-hKgILt jBOIA-D"
                opacity="0.3"
                size="163"
              />
            </div>
          </div>
          <a
            class="sc-iqHYGH cneMPT"
            font-size="3.4vh"
            height="4.25vh"
          >
            main with undefined points
          </a>
        </div>
        <div
          class="sc-iBPRYJ dRBUhu"
          width="NaNpx"
        >
          <header
            class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-1 MuiPaper-elevation4"
          >
            <div
              class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-2 MuiToolbar-gutters"
            >
              <span
                class="sc-bdfBwQ dYepCs"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-icon-3"
                  focusable="false"
                  title=""
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M21 6H3c-1.1 0-2 .9-2 2v8c0 1.1.9 2 2 2h18c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2zm-10 7H8v3H6v-3H3v-2h3V8h2v3h3v2zm4.5 2c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zm4-3c-.83 0-1.5-.67-1.5-1.5S18.67 9 19.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
                  />
                </svg>
                <h2
                  class="sc-gsTCUz"
                >
                  Additional
                </h2>
              </span>
            </div>
          </header>
          <span
            class="sc-dlfnbm bntuiM"
          >
            Keeped Piece
          </span>
          <div
            class="sc-eCssSg ciLaIA"
          >
            <div
              class="sc-jSgupP jQKSYs"
              height="15vh"
            />
          </div>
          <span
            class="sc-dlfnbm bntuiM"
          >
            Next Pieces
          </span>
          <div
            class="sc-fubCfw uDENn"
          >
            <div
              class="sc-eCssSg ciLaIA"
            >
              <div
                class="sc-jSgupP jQKSYs"
                height="15vh"
              >
                <div
                  class="sc-gKsewC eeibFv"
                >
                  <div
                    class="sc-hKgILt hBVxKa"
                    size="28"
                  />
                  <div
                    class="sc-hKgILt krEwUu"
                    size="28"
                  />
                  <div
                    class="sc-hKgILt hBVxKa"
                    size="28"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-dQppl hMUciP"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-crrsfI iVnQMr"
    >
      <div
        class="sc-pFZIQ gwVLIm"
      >
        <div
          class="sc-jrAGrp bNQYBx"
          height="80.75vhvh"
        >
          <div
            class="sc-kEjbxe kkVOZj"
          >
            <div
              class="sc-hKgILt jBOIA-D"
              opacity="0.3"
              size="163"
            />
          </div>
        </div>
        <a
          class="sc-iqHYGH cneMPT"
          font-size="3.4vh"
          height="4.25vh"
        >
          main with undefined points
        </a>
      </div>
      <div
        class="sc-iBPRYJ dRBUhu"
        width="NaNpx"
      >
        <header
          class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-1 MuiPaper-elevation4"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-2 MuiToolbar-gutters"
          >
            <span
              class="sc-bdfBwQ dYepCs"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-icon-3"
                focusable="false"
                title=""
                viewBox="0 0 24 24"
              >
                <path
                  d="M21 6H3c-1.1 0-2 .9-2 2v8c0 1.1.9 2 2 2h18c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2zm-10 7H8v3H6v-3H3v-2h3V8h2v3h3v2zm4.5 2c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zm4-3c-.83 0-1.5-.67-1.5-1.5S18.67 9 19.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
                />
              </svg>
              <h2
                class="sc-gsTCUz"
              >
                Additional
              </h2>
            </span>
          </div>
        </header>
        <span
          class="sc-dlfnbm bntuiM"
        >
          Keeped Piece
        </span>
        <div
          class="sc-eCssSg ciLaIA"
        >
          <div
            class="sc-jSgupP jQKSYs"
            height="15vh"
          />
        </div>
        <span
          class="sc-dlfnbm bntuiM"
        >
          Next Pieces
        </span>
        <div
          class="sc-fubCfw uDENn"
        >
          <div
            class="sc-eCssSg ciLaIA"
          >
            <div
              class="sc-jSgupP jQKSYs"
              height="15vh"
            >
              <div
                class="sc-gKsewC eeibFv"
              >
                <div
                  class="sc-hKgILt hBVxKa"
                  size="28"
                />
                <div
                  class="sc-hKgILt krEwUu"
                  size="28"
                />
                <div
                  class="sc-hKgILt hBVxKa"
                  size="28"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-dQppl hMUciP"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Game Filled Render 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-crrsfI iVnQMr"
      >
        <div
          class="sc-pFZIQ gwVLIm"
        >
          <div
            class="sc-jrAGrp bNQYBx"
            height="80.75vhvh"
          >
            <div
              class="sc-kEjbxe kkVOZj"
            >
              <div
                class="sc-hKgILt loolLL"
                opacity="1"
                size="163"
              />
            </div>
          </div>
          <a
            class="sc-iqHYGH cneMPT"
            font-size="3.4vh"
            height="4.25vh"
          >
            main with undefined points
          </a>
        </div>
        <div
          class="sc-dQppl hMUciP"
        >
          <div
            class="sc-pFZIQ gwVLIm"
          >
            <div
              class="sc-jrAGrp bjniTD"
              height="26.916666666666664vhvh"
            >
              <div
                class="sc-kEjbxe kkVOZj"
              >
                <div
                  class="sc-hKgILt fNEpTO"
                  opacity="0.3"
                  size="54"
                />
              </div>
            </div>
            <a
              class="sc-iqHYGH dIUbeX"
              font-size="1.1333333333333333vh"
              height="1.4166666666666665vh"
            >
              secondary with undefined points
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-crrsfI iVnQMr"
    >
      <div
        class="sc-pFZIQ gwVLIm"
      >
        <div
          class="sc-jrAGrp bNQYBx"
          height="80.75vhvh"
        >
          <div
            class="sc-kEjbxe kkVOZj"
          >
            <div
              class="sc-hKgILt loolLL"
              opacity="1"
              size="163"
            />
          </div>
        </div>
        <a
          class="sc-iqHYGH cneMPT"
          font-size="3.4vh"
          height="4.25vh"
        >
          main with undefined points
        </a>
      </div>
      <div
        class="sc-dQppl hMUciP"
      >
        <div
          class="sc-pFZIQ gwVLIm"
        >
          <div
            class="sc-jrAGrp bjniTD"
            height="26.916666666666664vhvh"
          >
            <div
              class="sc-kEjbxe kkVOZj"
            >
              <div
                class="sc-hKgILt fNEpTO"
                opacity="0.3"
                size="54"
              />
            </div>
          </div>
          <a
            class="sc-iqHYGH dIUbeX"
            font-size="1.1333333333333333vh"
            height="1.4166666666666665vh"
          >
            secondary with undefined points
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
